### PR TITLE
Add `ExcludeSelf` option for `Child` and `Parent` attributes

### PR DESCRIPTION
**Pull request by Niikelion**
Solves #19
Additionally it fixes defines (according to [unity documentation](https://docs.unity3d.com/Manual/PlatformDependentCompilation.html) you need to specify minor version too) as they were not functional.

### CODE REVIEW

**Comment by KyleBanks** [comment]
Hey, thanks for this!

Just a concern around the versions, Unity's documentation for version defines isn't super clear but the existing defines have been specifically added by community members on those versions to fix compilation. You're able to use any of the following formats (`UNITY_2020_OR_NEWER`, `UNITY_2020_3_OR_NEWER`, `UNITY_2020_3_1_OR_NEWER`, etc), only the C# version define requires major and minor. 

With this change we're only guarding against 2020.3+ whereas the version before included 2020.1 and 2020.2 which don't support the `FindObjectsByType(Type type, FindObjectsInactive findObjectsInactive, FindObjectsSortMode sortMode);` overload, but DO support `FindObjectsOfType(Type type, bool includeInactive);`. 

Basically with this change anyone on 2020.1 and 2020.2 will lose the `IncludeInactive` flag support.

Unfortunately I never set a minimum supported version and I know devs are using this on 2020 so I'd propose we instead do the following for now, and then officially set the minimum to 2021 in the future (probably when 2023 LTS releases):

```
#if UNITY_2020_3_OR_NEWER
                    `FindObjectsByType(Type type, FindObjectsInactive findObjectsInactive, FindObjectsSortMode sortMode);`
#elif UNITY_2020_OR_NEWER
                    ` FindObjectsOfType(Type type, bool includeInactive);`
#else
                    ` FindObjectsOfType(Type type);`
#endif
```

**Comment by Niikelion** [comment]
I changed  defines to `UNITY_2020_3_OR_NEWER` from `UNITY_2020_OR_NEWER` because I noticed newer versions of unity(2023) were using code from `#else` block same as old versions of unity. But I understand the issue and will add additional cases to make sure 2020.1 and 2020.2 are supported.

Screenshot_174
Screenshot_176

**Comment by KyleBanks** [comment]
Ah you're right, I just did a quick test and it's definitely always falling into the `#else` condition, I'm surprised that wasn't caught before. Thanks for this!

**Review by KyleBanks — APPROVED** [review]
